packages/vite/src/node/server/transformRequest.ts

```diff
@@ -249,10 +249,15 @@ async function loadAndTransform(
   }
   if (code == null) {
     const isPublicFile = checkPublicFile(url, config)
+    const publicDirName =
+      config.publicDir && config.root
+        ? path.relative(config.root, config.publicDir)
+        : '/public'
```

Comment:
`publicDir` and `root` should always exist, so you should be able to do this instead:
```suggestion
    let publicDirName = path.relative(config.root, config.publicDir)
    if (publicDirName[0] !== '.') publicDirName = '/' + publicDirName
```